docs/features/search/README.md

```diff
@@ -0,0 +1,100 @@
+---
+id: search-overview
+title: Search Documentation
+sidebar_label: Overview
+description:
+  Backstage Global Search lets you find the right information you are looking
+  for in the Backstage ecosystem.
+---
+
+# Backstage Global Search
+
+## What is it?
+
+Backstage Global Search lets you find the right information you are looking for
+in the Backstage ecosystem.
+
+## Features
+
+- A federated, faceted search, searching across all entities registered in your
+  Backstage instance.
+
+- A search that lets you plug in your own search engine of choice.
+
+- A standardized search API where plugin owners can choose to index their own
+  data.
+
+## Project roadmap
+
+| Version                 | Description                                                                                                                                                       |
+| ----------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Backstage Search V.0 ✅ | Search Frontend letting you search through the entities of the software catalog. [See V.0 Use Cases.](#backstage-search-v0)                                       |
+| Backstage Search V.1 ⌛ | Basic “out-of-the-box” in-memory indexing process of entities, and its metadata, registered to the Software Catalog. [See V.1 Use Cases.](#backstage-search-v1)   |
+| Backstage Search V.2 ⌛ | Search Backend responsible for the indexing process of entities, and its metadata, registered to the Software Catalog. [See V.2 Use Cases.](#backstage-search-v2) |
+| Backstage Search V.3 ⌛ | Standardized Search API lets you index your own data from your plugin to the search engine cluster of choice. [See V.3 Use Cases.](#backstage-search-v3)          |
+
+## Use Cases
+
+#### Backstage Search V.0
+
+- As a user I should be able to navigate to a search page and search for
+  entities registered in the Software Catalog.
+- As a user I should be able to use the search input field in the sidebar to
+  search for entities registered in the Software Catalog.
+- As a user I should be able to see the number of results my search returned.
+- As a user I should be able to filter on metadata (kind, lifecycle) when I’ve
+  performed a search.
+- As a user I should be able to hide the filters if I don’t need to use them.
```

Comment:
These can just be `User should be able to...`, but maybe not.

Reply:
This is leaning towards the traditional story outline of:

* As a `<user/persona>`...
* I should/would like to `<do this thing>`...
* So that I can `<obtain this result/value>`

I sort of like it as is, with the potential to add more reasoning for why a user would want to do any of those things at a later date? (the why helping the dev, esp open source ones, understand why they are doing what they are doing, other than someone else just gave them a TODO to blow through)

Reply:
Also to note, there may be some of those which may say "As a Backstage integrator, I should be able to..." as there may be features that relate to how to keep the search index healthy or understand how it's working, as opposed to just the end user as well?

Reply:
would say +1 to the above and yes, there is more potential to add more around the third bullet point.

Reply:
As a user
As a plugin developer
As a Backstage adopter
...
So many stories! :)

Makes sense to keep it. 👍

Reply:
@adamdmharvey Yeah we have some already saying "As a app developer" or "As a plugin developer", would you prefer "As a Backstage integrator" instead?

Reply:
@emmaindal I think so, only in so much as I've seen that being used a lot and I think it makes sense to me. I think to align it well with https://backstage.io/docs/overview/adopting, although even that page doesn't call it Integrator.

I do see this in the FAQ that lists three roles:
https://backstage.io/docs/FAQ#what-is-the-end-to-end-user-flow-the-happy-path-story

So maybe trying to keep all those consistent. (even if it means updating the FAQ, as "contributor" is also similar to your example of "plugin developer", etc.)

Reply:
@adamdmharvey Great thanks for linking that question from the FAQ and express your thinking!